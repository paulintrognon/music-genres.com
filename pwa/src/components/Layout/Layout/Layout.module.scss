.content {
  max-width: var(--max-width);
  min-height: 63vh;
  margin: auto;
  padding: 0 1rem;
  padding-bottom: 6rem;
  margin-top: 3rem;
  margin-bottom: 6rem;
}
