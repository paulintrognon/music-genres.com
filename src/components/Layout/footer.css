.main-footer {
  position: absolute;
  bottom: 0;
  width: 100%;
  height: 50px;
  line-height: 50px;
  text-align: center;
  font-weight: bold;
  font-size: 0.9rem;
  color: #fff;
}
