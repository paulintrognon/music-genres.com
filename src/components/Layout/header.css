.main-header {
  color: #fff;
  background-color: #00D1B2;
  box-shadow: 1px 0 3px #00D1B2;
  min-height: 100px;
  padding: 0 20px;
}

.main-nav .main-nav-brand {
  line-height: 100px;
  font-size: 3rem;
  text-transform: uppercase;
  font-family: 'fyodor';
  font-style: italic;
  text-shadow: 1px 1px 1px #007d6a;
}

.main-nav .side-links {
  float: right;
}

.main-nav .side-links .side-link {
  text-transform: uppercase;
  font-size: 1.2rem;
  line-height: 100px;
  padding: 20px;
  margin: 0 5px;
  transition: background-color 200ms, color 200ms;
}

.main-nav .side-links .side-link:hover {
  background-color: #05ffda;
  color: #008571;
}
