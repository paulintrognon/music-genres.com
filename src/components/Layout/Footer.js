import React from 'react';

import './footer.css';

export default function () {
  return (
    <footer className="main-footer">
      <a href="https://github.com/paulintrognon/" target="_blank" rel="noopener noreferrer">
        Paulin Trognon
      </a>
      &nbsp;-&nbsp;
      <a href="https://github.com/paulintrognon/music-genres.com" target="_blank" rel="noopener noreferrer">
        <i className="fa fa-github" aria-hidden="true"></i>
        &nbsp;Code source
      </a>
    </footer>
  )
}
