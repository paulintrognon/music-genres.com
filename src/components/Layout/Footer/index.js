import React from 'react';

import './footer.css';

import plusImage from '../../../images/plus.png';

export default class Footer extends React.Component {
  render() {
    return (
      <footer className="main-footer">
        <div className="main-footer-content">
          Design by&nbsp;
          <a href="https://fr.linkedin.com/in/margotbrun" target="_blank" rel="noopener noreferrer">
            Margot Brun
          </a>
          &nbsp;<img src={plusImage} className="plus-sign" alt="-" />&nbsp;
          <a href="https://github.com/paulintrognon/music-genres.com" target="_blank" rel="noopener noreferrer">
            Code by&nbsp;
          </a>
          <a href="https://paulintrognon.fr/" target="_blank" rel="noopener noreferrer">
            Paulin Trognon
          </a>
        </div>
      </footer>
    );
  }
}
