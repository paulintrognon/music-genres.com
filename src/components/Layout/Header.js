import React from 'react';
import { Link } from 'react-router-dom';

import './header.css';

export default function () {
  return (
    <header className="main-header">
      <nav className="main-nav">
        <Link to={'/'} className="main-nav-brand">music-genres.com</Link>

        <div className="side-links">
          <Link to={'/browse'} className="side-link">Browse</Link>
          <Link to={'/random'} className="side-link">Random</Link>
        </div>
      </nav>
    </header>
  )
}
