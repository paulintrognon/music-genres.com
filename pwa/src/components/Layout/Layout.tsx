import React from 'react'
import Footer from './Footer/Footer'
import Header from './Header/Header'
import classes from './Layout.module.scss'

type Props = {
  children: React.ReactNode
}

export const Layout: React.FunctionComponent<Props> = ({ children }) => {
  return (
    <div>
      <Header />
      <div className={classes.content}>{children}</div>
      <Footer />
    </div>
  )
}
