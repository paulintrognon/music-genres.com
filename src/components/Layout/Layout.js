import React from 'react';

import './layout.css';
import Header from './Header.js';
import Footer from './Footer.js';

export default class Layout extends React.Component {
  render() {
    return (
      <div>
        <Header></Header>
        <div className="content-container">
          {this.props.children}
        </div>
        <Footer></Footer>
      </div>
    )
  }
}
