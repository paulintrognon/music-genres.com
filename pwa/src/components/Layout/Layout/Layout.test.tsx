import { Layout } from './Layout'
import { shallow } from 'enzyme'

it('should render Layout', () => {
  const wrapper = shallow(<Layout>Hello world</Layout>)
  expect(wrapper).toMatchSnapshot()
})
