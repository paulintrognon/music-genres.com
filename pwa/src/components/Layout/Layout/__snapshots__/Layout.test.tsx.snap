// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Layout 1`] = `
<div>
  <Header />
  <div
    className="content"
  >
    Hello world
  </div>
  <Footer />
</div>
`;
